contracts/commands/BasicBuyCommand.sol

```diff
@@ -126,22 +125,14 @@ contract BasicBuyCommand is BaseMPACommand, Ownable {
         returns (bool)
     {
         BasicBuyTriggerData memory decoded = decode(triggerData);
-        (uint256 collRatio, uint256 nextCollRatio, , bytes32 ilk) = getVaultAndMarketInfo(cdpId);
 
-        SpotterLike spot = SpotterLike(serviceRegistry.getRegisteredService(MCD_SPOT_KEY));
-        (, uint256 liquidationRatio) = spot.ilks(ilk);
+        McdView mcdView = McdView(serviceRegistry.getRegisteredService(MCD_VIEW_KEY));
+        uint256 nextCollRatio = mcdView.getRatio(cdpId, true);
 
         (uint256 lowerTarget, uint256 upperTarget) = decoded.targetCollRatio.bounds(
             decoded.deviation
         );
 
-        bool isWithinBounds = nextCollRatio <= upperTarget.wad() &&
-            nextCollRatio >= lowerTarget.wad();
-        bool isNearLiquidationRatio = collRatio < nextCollRatio &&
-            collRatio * 10**9 <
-            liquidationRatio.mul(RatioUtils.RATIO + liquidationRatioPercentage).div(
-                RatioUtils.RATIO
-            );
-        return isWithinBounds || isNearLiquidationRatio;
+        return nextCollRatio <= upperTarget.wad() && nextCollRatio >= lowerTarget.wad();
```

Comment:
I guess we still need here 

&& lowerTarget > liquidationRatio 

otherwise we are hoping for correct values from UI

Reply:
you are commenting on is execution correct, that check should not be here